notification/src/api/notification/notification.controller.js

```diff
@@ -1,30 +1,97 @@
 const _ = require('lodash')
 const httpUtils = require('../../utils/commons')
-const ctp = require('../../utils/ctp')
 const {
-  processNotifications,
+  processNotification,
 } = require('../../handler/notification/notification.handler')
-const config = require('../../config/config')()
+const config = require('../../config/config')
 const logger = require('../../utils/logger').getLogger()
 
-const ctpClient = ctp.get(config)
+class ValidationError extends Error {
+  constructor({ stack, message, notification, isRecoverable }) {
+    super()
+    this.stack = stack
+    this.message = message
+    this.notification = JSON.stringify(notification)
+    this.isRecoverable = isRecoverable
+  }
+}
 
 // TODO: add JSON schema validation:
 // https://github.com/commercetools/commercetools-adyen-integration/issues/9
 async function handleNotification(request, response) {
   if (request.method !== 'POST') return httpUtils.sendResponse(response)
   const body = await httpUtils.collectRequestData(request)
   try {
-    const notification = _.get(JSON.parse(body), 'notificationItems', [])
-    await processNotifications(notification, ctpClient)
+    const notifications = _.get(JSON.parse(body), 'notificationItems', [])
+    for (const notification of notifications) {
+      const ctpProjectConfig = getCtpProjectConfig(notification)
+      const adyenConfig = getAdyenConfig(notification)
+
+      await processNotification(
+        notification,
+        adyenConfig.enableHmacSignature,
+        ctpProjectConfig
+      )
+    }
     return sendAcceptedResponse(response)
   } catch (err) {
     logger.error(
       { adyenRequestBody: `${body}`, err },
       'Unexpected exception occurred.'
     )
-    return httpUtils.sendResponse(response, 500)
+    if (err.isRecoverable) {
```

Comment:
What is the impact on Adyen's side, should we return 500 ? or accepted for every case? 
When Adyen stops sending notifications?

Reply:
If we accept 200, they will stop sending notifications. I was thinking that in case notification doesn't have a project configured, it should return 500 so that Adyen will keep on retry. It could be that one forget to configure, so at least this will give them a chance to reconfigure and fix that.

Reply:
I feel that we need to consult Adyen for best practices here, can you ask them ?

So in the basics when the module acknowledges the notification, Adyen servers will not send it back. 
In docs says: https://docs.adyen.com/development-resources/webhooks#accept-notifications
if the Adyen server is not receiving `[accepted]` from the integration it will block all notifications (looks like queued)
https://docs.adyen.com/development-resources/webhooks/best-practices#queued-notifications so is it good to cascade notifications on the queue until the devs configured it well (does not feel right on the production use case).

Reply:
I will ask them but I remember we asked them many times in the past and they always say accept it and it was not very helpful 😄